curated_transformers/layers/attention.py

```diff
@@ -407,6 +546,11 @@ def forward(
             )
 
         if _TORCH_SDP.get():
+            if self.use_alibi:
+                msg = "ALiBi disabled when using PyTorch SDP"
+                warnings.filterwarnings(action="once", message=msg)
+                warnings.warn(msg)
+
```

Comment:
I think this should be an exception, since it would return incorrect outputs. However, I think we could support ALiBi with Torch SDP. The `attn_mask` argument of `F.scaled_dot_product_attention` is additive when it's a float tensor. ALiBi is also additive, so we could make it work by adding the `ALiBi` biases to the logit mask for positions that are not masked out.

Reply:
Added.